commerce_coordinator/apps/demo_lms/pipeline.py

```diff
@@ -0,0 +1,31 @@
+"""
+Demo LMS app filter pipeline implementations.
+"""
+
+from openedx_filters import PipelineStep
+
+
+# PipelineSteps augment, filter, or transform data passed into them and return a dict that gets merged into the
+# `**kwargs` passed to the next step in the pipeline.  The merge is done using
+# [dict.update()](https://docs.python.org/3/library/stdtypes.html#dict.update)
+#
+# By overriding `run_filter` and adding expected arguments to the signature, we get a small amount checking that the
+# name, for example here "sample_data", was among the arguments passed in.
+class SomeData(PipelineStep):
+    """
+    Adds data to the sample data list.
+    """
+    def run_filter(self, sample_data, *args, **kwargs):  # pylint: disable=arguments-differ
+        return {
+            "sample_data": sample_data + ["A string", {"dict": "like me"}],
+        }
```

Comment:
To confirm, if we were doing the Order History example, we would have an OrderHistory class (here or wherever we were saving our pipeline) and then within its run_filter method that's where we'd call out to oscar or commercetools, is that right?

Reply:
Correct.